osf/metrics.py

```diff
@@ -175,3 +175,99 @@ class PreprintView(BasePreprintMetric):
 
 class PreprintDownload(BasePreprintMetric):
     pass
+
+
+class UserInstitutionProjectCounts(MetricMixin, metrics.Metric):
+    user_id = metrics.Keyword(index=True, doc_values=True, required=True)
+    institution_id = metrics.Keyword(index=True, doc_values=True, required=True)
+    # TODO: Uncomment following line and include department in metrics when available as a OSFUser attribute
+    # department = metrics.Keyword(index=True, doc_values=True, required=False)
+    public_project_count = metrics.Integer(index=True, doc_values=True, required=True)
+    private_project_count = metrics.Integer(index=True, doc_values=True, required=True)
+
+    class Index:
+        settings = {
+            'number_of_shards': 1,
+            'number_of_replicas': 1,
+            'refresh_interval': '1s',
+        }
+
+    @classmethod
+    def record_user_institution_project_counts(cls, user, institution, public_project_count, private_project_count, **kwargs):
+        return cls.record(
+            user_id=user._id,
+            institution_id=institution._id,
+            # TODO: Uncomment following line and include department in metrics when available as a OSFUser attribute
+            # department=user.department,
+            public_project_count=public_project_count,
+            private_project_count=private_project_count,
+            **kwargs
+        )
+
+    @classmethod
+    def get_latest_user_institution_project_counts(cls, user, institution):
+        search = cls.search().filter('match', user_id=user._id).filter('match', institution_id=institution._id).sort('-timestamp')[:1]
+
+        try:
+            response = search.execute()
+        except NotFoundError:
+            # _get_relevant_indices returned 1 or more indices
```

Comment:
We use this exact span of code with these exact comments 5 fives in the repo, maybe make this a util? or remove 213-117, because I'm not sure this error can actually happen in this context.